scipy/stats/_continuous_distns.py

```diff
@@ -8993,6 +8993,10 @@ def _pdf(self, x, kappa):
         #                        = exp(kappa * cosm1(x)) / (2*pi*i0e(kappa))
         return np.exp(kappa*sc.cosm1(x)) / (2*np.pi*sc.i0e(kappa))
 
+    def _logpdf(self, x, kappa):
+        # vonmises.pdf(x, kappa) = exp(kappa * cos(x)) / (2*pi*I[0](kappa))
+        return kappa * np.cos(x) - np.log(2*np.pi) - np.log(sc.i0(kappa))
```

Comment:
Why not base this off the formulation of `pdf` used above?
```suggestion
        return kappa*sc.cosm1(x) - np.log(2*np.pi) - np.log(sc.i0e(kappa))
```
It avoids the `sc.i0` overflows.

Reply:
@mdhaber Indeed this is much better. Thanks!
Sorry this is my first PR for scipy so I am a bit confused with the organization of the test code. Could you suggest a place where I can add this test function?

Reply:
Ah never mind, I think I figured it out.

Reply:
Just updated the code.